java/ql/src/Security/CWE/CWE-502/UnsafeDeserialization.qhelp

```diff
@@ -51,6 +51,56 @@ from the input stream removes the vulnerability.
 
 </example>
 
+<p>
+
+Fixes by framework
+<table>
+    <tbody>
+        <tr>
+            <th>Project</th>
+            <th>Maven Cordinates</th>
+            <th>Secure by Default</th>
+            <th>Fix</th>
+        </tr>
+            <tr>
+            <td>XMLDecoder</td>
+            <td>Java Standard Library</td>
+            <td>No</td>
+            <td>Don't use XMLDecoder with untrusted user input. It is impossible to secure.</td>
+        </tr>
+        <tr>
+            <td>ObjectInputStream</td>
+            <td>Java Standard Library</td>
+            <td>No</td>
+            <td>Leverage a validating input stream like <code>org.apache.commons.io.serialization.ValidatingObjectInputStream</code></td>
+        </tr>
+            <tr>
+            <td>FastJson</td>
+            <td>com.alibaba:fastjson</td>
+            <td>Partially</td>
+            <td>Call <code>com.alibaba.fastjson.parser.ParserConfig#setSafeMode</code> with the argument <code>true</code></td>
+        </tr>
+        <tr>
+            <td>SnakeYAML</td>
+            <td>org.yaml:snakeyaml</td>
+            <td><a href="https://bitbucket.org/snakeyaml/snakeyaml/wiki/CVE%20&amp;%20NIST.md">No</a>. <a href="https://bitbucket.org/snakeyaml/snakeyaml/issues/561/cve-2022-1471-vulnerability-in">Maintainer response</a>.</td>
+            <td>Instantiate the <code>org.yaml.snakeyaml.Yaml</code> instance explicitly with an instance of <code>org.yaml.snakeyaml.constructor.SafeConstructor</code> as an argument.</td>
+        </tr>
```

Comment:
This has been my pet project the past two weeks. I'm still working to convince the maintainer to make this secure by default. If this does end up happening, I'll update this documentation.